### PR TITLE
Remove some InstanceType fields to simplify events

### PR DESCRIPTION
**Pull request by alexandra-bucur**
Signed-off-by: Alexandra Bucur <alexandra-nona.bucur@ing.com>

## Description

In order to improve performance when it comes to events sent through the cohort, I removed typeDefDescription, typeDefDescriptionGUID, typeDefSuperTypes, validStatusList and validInstanceProperties fields from InstanceType class.

## Testing

JUnits, FVTs, CTS repository workbench, Notebook labs for building a data catalog - passed.

### CODE REVIEW

File: open-metadata-implementation/framework-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/converters/ElementHeaderConverter.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public void updateBean(ElementBase bean)
     {
         if (entity != null)
         {
-            TypeConverter typeConverter = new TypeConverter();
+            TypeConverter typeConverter = new TypeConverter(repositoryHelper, serviceName);
```
[the remark this answers is not in the document]
Comment (alexandra-bucur) [reply open-metadata-implementation/framework-services/ocf-metadata-management/ocf-metadata-handlers/src/main/java/org/odpi/openmetadata/commonservices/ocf/metadatamanagement/converters/ElementHeaderConverter.java:104]: Coding style rule not agreed within community.

File: open-metadata-implementation/access-services/security-officer/security-officer-server/src/main/java/org/odpi/openmetadata/accessservices/securityofficer/server/admin/SecurityOfficerAdmin.java [hunk 2]
```diff
@@ -57,13 +58,17 @@ public synchronized void initialize(AccessServiceConfig accessServiceConfigurati
         try
         {
             this.auditLog = auditLog;
+            String accessServiceName = accessServiceConfigurationProperties.getAccessServiceName();
             List<String> supportedZones = super.extractSupportedZones(accessServiceConfigurationProperties.getAccessServiceOptions(),
-                                                                      accessServiceConfigurationProperties.getAccessServiceName(), auditLog);
+                    accessServiceName, auditLog);
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/security-officer/security-officer-server/src/main/java/org/odpi/openmetadata/accessservices/securityofficer/server/admin/SecurityOfficerAdmin.java:63]: Indent so all parameters are aligned

File: open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java [hunk 3]
```diff
@@ -254,6 +254,21 @@ public TypeDef getTypeDefByName(String sourceName,
...
 
+    /**
+     * Gets super types for given type name.
+     *
+     * @param sourceName  the source of the request (used for logging)
+     * @param typeDefName unique name for the TypeDef
+     * @return the super types
+     */
+    @Override
+    public List<TypeDefLink> getSuperTypes(String sourceName, String typeDefName) {
```
Comment (mandy-chessell) [inline open-metadata-implementation/repository-services/repository-services-implementation/src/main/java/org/odpi/openmetadata/repositoryservices/localrepository/repositorycontentmanager/OMRSRepositoryContentHelper.java:265]: Braces on new line

**Comment by cmgrote** [comment]
Is this considered a backwards-compatible change? (Or maybe it doesn't need to be given we're bumping to v4.0?)

This has an impact on repository connectors that are, for example, storing all of these details as part of the instance type. (And in cases like XTDB relying on them to do validation of the details of instances — supertypes, valid properties, etc.)

**Comment by alexandra-bucur** [comment]
@cmgrote Hi Chris! Let's discuss this on the next developers call.

**Pull request merged**